src/main/java/uk/ac/ebi/atlas/hcalandingpage/HcaHumanExperimentDao.java

```diff
@@ -18,41 +22,46 @@
 import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.ONTOLOGY_ANNOTATION_ANCESTORS_URIS;
 import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.ONTOLOGY_ANNOTATION_LABEL;
 
-@Component
+@Repository
 public class HcaHumanExperimentDao {
     private final SingleCellAnalyticsCollectionProxy singleCellAnalyticsCollectionProxy;
 
     public HcaHumanExperimentDao(SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory) {
         this.singleCellAnalyticsCollectionProxy = solrCloudCollectionProxyFactory.create(SingleCellAnalyticsCollectionProxy.class);
     }
 
-    public ImmutableSet<String> fetchExperimentAccessions(String characteristicName, String characteristicValue) {
+    public ImmutableSet<String> fetchExperimentAccessions(String characteristicName,
+                                                          Set<String> characteristicValues) {
         var queryBuilder = new SolrQueryBuilder<SingleCellAnalyticsCollectionProxy>();
         /* Reason for checking blank here is in case characteristicValue is empty our solrQueryBuilder
          * builds query like characteristic_value:("\*") which is different from characteristic_value:*
          * which we need. So to handle that situation if characteristicValue is empty we build query
          * characteristic_name:("organism_part") as we need all the experiments in case of empty characteristicValue
          * */
-        queryBuilder
-                .setNormalize(false)
-                .addQueryFieldByTerm(CHARACTERISTIC_NAME, characteristicName)
+
+        queryBuilder.addQueryFieldByTerm(CHARACTERISTIC_NAME, characteristicName)
```

Comment:
We need the normalize flag to exist for the cases when we just pass ontology id `e.g. UBERON_0000029`

When the normalize flag is not set then the resulting solr query is not correctly formed, which in this case would be `q=characteristic_name:("organism_part") AND characteristic_value("UBERON_0000029")...` which would return empty result set as there is no field with `characteristic_value: UBERON_0000029`.

So with the normalizing flag set to `false` the resulting query would be `q=characteristic_name:("organism_part") AND characteristic_value:*UBERON_0000029 ....` which is what we need.

Reply:
Yes, it make sense. I fixed it.